### ISSUE TITLE
Build fails with: could not determine kind of name for C.mrb_ary_len

### ISSUE THREAD

**Issue by cromega** [mitchellh/go-mruby#72]
Hi,

Building the library fails as follows:

```
go-mruby git:(master) go build
# github.com/mitchellh/go-mruby
./array.go:15:13: could not determine kind of name for C.mrb_ary_len
```

A commit to MRuby from 2017 (`67e2dddb8254fbf4a4eb81c1241ec1963bc77627`) seems to have introduced the following change:

```
    Remove `mrb_ary_len` function. Use `RARRAY_LEN` instead.
```

Wondering why it hasn't caused problems for anyone else. Am I doing something stupid?

Go: 1.11.5
Mruby: 2.0.0

**Comment by cromega** [comment]
I actually came up with a pretty simple workaround for this issue, sadly there are other errors so I couldn't run the tests yet.

Is it a case of the code not being compatible with more recent versions of mruby and it's probably not going to work?

**Comment by erikh** [comment]
Yes; as of 1.4 or so the build doesn't work with the mruby source anymore.
There's an open ticket for it (not near github atm).

**Comment by erikh** [comment]
If you leverage the 1.2.0 tag, it will all work.

**Comment by erikh** [comment]
see #66 for more information.

**Comment by wade-welles** [comment]
Still getting this for the last 6 months

**Comment by erikh** [comment]
The comment right above yours tells you how to resolve it. I'm genuinely
sorry I haven't had time to completely rewrite the C layer (which is what
is required), but you're welcome to send patches if it bothers you so.

**Comment by wade-welles** [comment]
I have tried in that in the past without success unfortunately. I will build a new VM and try again. Id be happy to rewrite it, I'm juggling a few things so any pointers or notes you have that could help me get started quicker would be appreciated. 

I was considering implementing a lexer and parser and finishing off GoRuby so the community has something, but obviously that's not the most ideal route since it is implementing it without a VM. The quickest solution seems obviously to carry on the embedding, I have even begun experimenting with different ways of embedding the binary so that it will directly use the executable (i.e. `/usr/bin/mruby` or just `ruby`).
